### PR TITLE
Restrict users from using certain commands for Bookmark Transactions

### LINKED ISSUE #167: Restrict users from using certain commands for Frequent Transactions
Currently. user can utilise edit, delete and convert frequent incomes and expenses from any tab. This might not be user-friendly as the user will not be able to see which frequent income and expense they want to execute the respective command on.

To restrict the edit, delete and convert commands, for both `Frequent Incomes` and `Frequent Expenses`, by creating a generic edit and delete command which will be dependent on the `UiState`.

### PR DESCRIPTION
**Pull request by wltan**
Resolves #167 

Editing/Deleting/Converting bookmark expenses/incomes have been changed to use a generic command based on the current tab.

Adding bookmark expenses/incomes remain unchanged.

User guide has been updated to reflect the new behavior and also be more consistent with the other sections (although there is probably still room to improve)

Outstanding tasks: ~~Update user guide to reflect the new commands (hopefully within this PR)~~ (done), test coverage (possibly pushed to v1.4)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -465,286 +465,238 @@ Expected Outcome:
...
-The inputted amount can have either 0 or 2 decimal places and also have an optional $ prefix.
+- The title, amount, and categories of the bookmark expense follows the same rules as the [Add Expense](#431-add-expense) feature above.
 
->  :bulb: Inputting of categories for the bookmark expense is optional and is up to the user's discretion.
+Shortcut: `addbe`
 
-**Examples**:
+Examples:
 - `add-bookmark-expense t/Phone Bill a/60 c/Utilities c/Personal`
+adds a bookmark expense titled `Phone Bill` with amount `$60.00` and two categories`Utilities` and `Personal`.
```
Comment (ianyong) [inline docs/UserGuide.md:483]:
```suggestion
adds a bookmark expense titled `Phone Bill` with amount `$60.00` and two categories `Utilities` and `Personal`.
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -465,286 +465,238 @@ Expected Outcome:
...
 - `add-bookmark-expense t/Spotify Subscription a/$9 c/Others`
+adds a bookmark expense titled `Spotify Subscription` with amount `$9.00` and a single category `Others`.
 - `addbe t/Bubble Tea a/$4.50 c/Food & Drink`
+adds a bookmark expense titled `Bubble Tea` with amount `$4.50` and a single category `Food & Drink`.
```
Comment (ianyong) [inline docs/UserGuide.md:487]: Should we standardise `Food & Drink` to `Food & Beverage`?

**Review by ianyong — CHANGES_REQUESTED** [review]

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by siddarth2824 — APPROVED** [review]
LGTM!

**Pull request merged**